### ISSUE TITLE
Tell meson to not search for CCache

### ISSUE THREAD

**Issue by Salamandar** [mesonbuild/meson#2991]
When building on ArchLinux with `!ccache` in makepkg.conf, Meson still searches for CCache.
Meson could expose a "--no-ccache" argument to disable the automatic CCache search, even if CC/CXX is *not* in the environment.

Yes, I'm aware of https://github.com/mesonbuild/meson/pull/1506 but we need to define CC/CXX, something that Makepkg doesn't do.

**Comment by jpakkane** [comment]
Is the problem that it searches for ccache or the fact that it finds it? If the latter then your build env should make sure that ccache is not in path when that option is set.

**Comment by Salamandar** [comment]
I'd say the problem is that Meson *searches* for CCache. If `!ccache` is in makepkg.conf, that means the build script should not even search for CCache.

But we don't want to "shadow" ccache from the env: what if we just want to detect the available CCache version, without actually using it ? (when building an IDE, plugin, build system,...)

**Comment by 1ace** [comment]
Just ran into this issue; is there any workaround until `--ccache=false` gets implemented?

**Comment by nirbheek** [comment]
Maybe try setting `ccache = 'false'` in the native file?

**Comment by eli-schwartz** [comment]
> Is the problem that it searches for ccache or the fact that it finds it? If the latter then your build env should make sure that ccache is not in path when that option is set.

This is a bit difficult to do when the binaries for *everything* are all in /usr/bin -- it's of course doable when building every software compilation in a clean build chroot with minimal dependencies.

Traditionally, the Arch Linux build system does not care whether ccache is installed, but if the buildenv is told to *use* ccache, it will prepend /usr/lib/ccache/bin to the $PATH in order to shadow the compiler executables.

**Comment by henning-schild** [comment]
`export CCACHE_DISABLE=1` now the wrapper will be a noop, just like it was never detected in the first place

**Comment by intelfx** [comment]
Same issue exists with sccache. Meson prefers sccache to ccache, however 1) sccache is slower and 2) sccache is significantly less flexible/configurable. For me, this behavior of Meson is actually counterproductive, _and_ you cannot disable/override it.

**Comment by jpakkane** [comment]
If you set your compiler either with a native/cross file or with an environment variable, then Meson will use that and won't add ccache/sccache even if they exist in path.

**Comment by Salamandar** [comment]
@jpakkane a cross file is clearly more difficult to set up than to add a command line flag when calling meson.

**Comment by eli-schwartz** [comment]
I'm decidedly unsure how that answers what @jpakkane said?

The suggestion was "either a native file or an environment variable", and an environment variable and a command line flag are equally easy to set up, differing only in whether it comes before or after the `meson` command itself.

**Comment by Salamandar** [comment]
Yes, indeed, an environment variable is easy to do. I misread the message.

**Comment by lilydjwg** [comment]
Hello, what's the current way to disable ccache for meson? I _guess_ that `export CCACHE_DISABLE=1` worked, but I'm not sure since ccache is still invoked.

**Comment by dcbaker** [comment]
@lilydjwg that does work. You can also use `CC=<my complier>` (or CXX=...`), in which case Meson wont add ccache

**Comment by eli-schwartz** [comment]
Keep in mind that meson's handling of ccache is part of meson's ***automatic compiler detection***. It's the same thing that makes meson search for any of `cc`, `gcc`, `clang`, `nvc`, `pgcc`, `icc`, `icx` and arbitrarily guess which one you probably want.

Setting $CC will always tell meson ***what you want***, and if $CC says "gcc" then meson will use "gcc" and not use "ccache gcc".

**Comment by lilydjwg** [comment]
Thanks!
